alexk_low_cost_robot/wizard/src/main.rs

```diff
@@ -0,0 +1,158 @@
+use std::time::Duration;
+use clap::Parser;
+use rustypot::device::{xl330, xl430};
+use rustypot::DynamixelSerialIO;
+use serialport::SerialPort; // 4.2.7
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+pub struct Cli {
+    #[clap(
+        long, value_delimiter = ' ', num_args = 1.., default_value = "false false false false false false"
+    )]
+    pub inverted: Vec<bool>,
+
+    #[clap(long, action)]
+    pub homing: bool,
+
+    #[clap(long, action)]
+    pub drive: bool,
+
+    #[clap(long)]
+    pub port: String,
+
+    #[clap(long, action)]
+    pub puppet: bool,
+}
+
+
+fn convert_to_extended_mode(angle: u32) -> i32 {
+    if angle < i32::MAX as u32 {
+        angle as i32
+    } else {
+        -((u32::MAX - angle) as i32)
+    }
+}
+
+fn get_positions(io: &DynamixelSerialIO, serial_port: &mut dyn SerialPort, puppet: bool) -> Vec<i32> {
+    let pos = if !puppet {
+        let pos = xl330::sync_read_present_position(
+            &io,
+            serial_port,
+            &[0, 1, 2, 3, 4, 5],
+        ).expect("Read Communication error");
+
+        pos.iter().map(|&x| convert_to_extended_mode(x)).collect::<Vec<_>>()
```

Comment:
Could be good to reference some of the Robotis documentation for things like extended mode that might not be obvious for everyone.

Reply:
Yeah I'll do some doc on the code to be more explicit about what are each parameter